src/parser.rs

```diff
@@ -54,204 +50,242 @@ macro_rules! parse_num {
             }
         }
+        _ => unreachable!(),
+    })
+}
+// rhs = { str_literal | ip_literal | int_literal }
+fn parse_rhs(pair: Pair<Rule>) -> ParseResult<Value> {
+    let pairs = pair.into_inner();
+    let pair = pairs.peek().unwrap();
+    let rule = pair.as_rule();
+    Ok(match rule {
+        Rule::str_literal => Value::String(parse_str_literal(pair)?),
+        Rule::ip_literal => Value::IpCidr(parse_ip_literal(pair)?),
+        Rule::int_literal => Value::Int(parse_int_literal(pair)?),
+        _ => unreachable!(),
+    })
+}
 
-        Ok(s)
-    }
-
-    fn ipv4_cidr_literal(input: Node) -> ParseResult<Ipv4Cidr> {
-        input.as_str().parse().into_parse_result(&input)
-    }
-
-    fn ipv6_cidr_literal(input: Node) -> ParseResult<Ipv6Cidr> {
-        input.as_str().parse().into_parse_result(&input)
-    }
-
-    fn ipv4_literal(input: Node) -> ParseResult<Ipv4Cidr> {
-        format!("{}/32", input.as_str())
-            .parse()
-            .into_parse_result(&input)
-    }
-
-    fn ipv6_literal(input: Node) -> ParseResult<Ipv6Cidr> {
-        format!("{}/128", input.as_str())
-            .parse()
-            .into_parse_result(&input)
-    }
-
-    fn ip_literal(input: Node) -> ParseResult<IpCidr> {
-        Ok(match_nodes! { input.children();
-            [ipv4_cidr_literal(c)] => IpCidr::V4(c),
-            [ipv6_cidr_literal(c)] => IpCidr::V6(c),
-            [ipv4_literal(c)] => IpCidr::V4(c),
-            [ipv6_literal(c)] => IpCidr::V6(c),
-        })
-    }
-
-    fn int_literal(input: Node) -> ParseResult<i64> {
-        use Rule::*;
-
-        let digits_node = input.children().single().unwrap();
-
-        let radix = match digits_node.as_rule() {
-            hex_digits => 16,
-            oct_digits => 8,
-            dec_digits => 10,
+// str_literal = ${ "\"" ~ str_inner ~ "\"" }
+fn parse_str_literal(pair: Pair<Rule>) -> ParseResult<String> {
+    let char_pairs = pair.into_inner();
+    let mut s = String::new();
+    for char_pair in char_pairs {
+        let rule = char_pair.as_rule();
+        match rule {
+            Rule::str_esc => s.push(parse_str_esc(char_pair)),
+            Rule::str_char => s.push(parse_str_char(char_pair)),
             _ => unreachable!(),
-        };
-
-        let mut num = parse_num!(digits_node, i64, radix)?;
-
-        if let Some('-') = input.as_str().chars().next() {
-            num = -num;
         }
-
-        Ok(num)
     }
+    Ok(s)
+}
 
-    fn rhs(input: Node) -> ParseResult<Value> {
-        Ok(match_nodes! { input.children();
-            [str_literal(s)] => Value::String(s),
-            [ip_literal(ip)] => Value::IpCidr(ip),
-            [int_literal(i)] => Value::Int(i),
-        })
+fn parse_str_esc(pair: Pair<Rule>) -> char {
+    let pairs = pair.into_inner();
+    match pairs.as_str() {
+        "\\\"" => '"',
+        "\\\\" => '\\',
+        "\\n" => '\n',
+        "\\r" => '\r',
+        "\\t" => '\t',
```

Comment:
It's hard to read, could we change them to raw string?

Reply:
updated